integration/devmode/devmode_test.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright IBM Corp All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package devmode
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"syscall"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/hyperledger/fabric/integration/nwo"
+	"github.com/hyperledger/fabric/integration/nwo/commands"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	"github.com/tedsuo/ifrit"
+	"github.com/tedsuo/ifrit/ginkgomon"
+)
+
+var _ = Describe("Devmode", func() {
+	var (
+		testDir                  string
+		client                   *docker.Client
+		network                  *nwo.Network
+		process                  ifrit.Process
+		chaincode                nwo.Chaincode
+		chaincodeRunner          *ginkgomon.Runner
+		chaincodeProcess         ifrit.Process
+		channelName, chaincodeID string
+	)
+
+	BeforeEach(func() {
+		var err error
+		channelName = "testchannel"
+		testDir, err = ioutil.TempDir("", "devmode")
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err = docker.NewClientFromEnv()
+		Expect(err).NotTo(HaveOccurred())
+
+		chaincode = nwo.Chaincode{
+			Name:    "mycc",
+			Version: "0.0",
+			Path:    "github.com/hyperledger/fabric/integration/chaincode/simple/cmd",
+			Ctor:    `{"Args":["init","a","100","b","200"]}`,
+			Policy:  `AND ('Org1MSP.member','Org2MSP.member')`,
+		}
+
+		chaincodeID = fmt.Sprintf("%s:%s", chaincode.Name, chaincode.Version)
+		network = nwo.New(nwo.BasicSolo(), testDir, client, BasePort(), components)
+		network.TLSEnabled = false
+		// turn on dev mode for all peers
+		for _, p := range network.Peers {
+			p.DevMode = true
+		}
+
+		network.GenerateConfigTree()
+		network.Bootstrap()
+
+		networkRunner := network.NetworkGroupRunner()
+		process = ifrit.Invoke(networkRunner)
+		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+
+	AfterEach(func() {
+		By("Cleanup")
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+
+		if chaincodeProcess != nil {
+			chaincodeProcess.Signal(syscall.SIGTERM)
+		}
+
+		if network != nil {
+			network.Cleanup()
+		}
+
+		os.RemoveAll(testDir)
+	})
+
+	It("executing chaincode in dev mode", func() {
```

Comment:
```suggestion
	It("executes chaincode in dev mode", func() {
```
Nit: these should read as a sentence: "It executes chaincode in devmode" (applies here and below)

Reply:
🧹